src/legacy/server/saved_objects/service/lib/repository.d.ts

```diff
@@ -26,7 +26,22 @@ export interface SavedObjectsRepositoryOptions {
   schema: unknown;
   serializer: unknown;
   migrator: unknown;
-  onBeforeWrite: unknown;
+  onBeforeWrite?: (
+    action: 'update' | 'bulk' | 'delete' | 'deleteByQuery',
```

Comment:
I think it's missing `create` and `index` which are passed in from `create` function.